app/ts/AddressBook.tsx

```diff
@@ -167,57 +167,70 @@ function AddressBookEntryCard({ removeEntry, renameAddressCallBack, ...entry }:
 type ViewFilter = {
 	activeFilter: FilterKey
 	searchString: string
-	rpcNetwork: { readonly name: string, readonly chainId: bigint } | undefined
+	chain: ChainEntry | undefined
+}
+
+type AddressBookEntriesWithFilter = {
+	addressBookEntries: AddressBookEntries
+	activeFilter: FilterKey
 }
 
 export function AddressBook() {
-	const addressBookEntries = useSignal<AddressBookEntries>([])
-	const currentRpcNetwork = useSignal<RpcNetwork | undefined>(undefined)
+	const addressBookEntriesWithFilter = useSignal<AddressBookEntriesWithFilter>({ addressBookEntries: [], activeFilter: 'My Active Addresses' })
+	const addressBookEntries = useComputed(() => addressBookEntriesWithFilter.value.addressBookEntries || [])
+	const currentChain = useSignal<ChainEntry | undefined>(undefined)
+	const currentChainId = useComputed(() => currentChain.value?.chainId || 1n)
 	const rpcEntries = useSignal<RpcEntries>([])
-	const viewFilter = useSignal<ViewFilter>({ activeFilter: 'My Active Addresses', searchString: '', rpcNetwork: undefined })
+	const viewFilter = useSignal<ViewFilter>({ activeFilter: 'My Active Addresses', searchString: '', chain: undefined })
 	const modalState = useSignal<Modals>({ page: 'noModal' })
+	const modifyAddressSignal = useComputed(() => modalState.value.page === 'addNewAddress' ? modalState.value.state : undefined)
+	function sendQuery() {
+		const filterValue = viewFilter.value
+		if (filterValue.chain === undefined) return
+		sendPopupMessageToBackgroundPage({ method: 'popup_getAddressBookData', data: {
+			chainId: filterValue.chain.chainId,
+			filter: filterValue.activeFilter,
+			searchString: filterValue.searchString
+		} })
+	}
+	useSignalEffect(sendQuery)
+
 	useEffect(() => {
-		const popupMessageListener = async (msg: unknown) => {
+		const popupMessageListener = (msg: unknown) => {
 			const maybeParsed = MessageToPopup.safeParse(msg)
 			if (!maybeParsed.success) return // not a message we are interested in
 			const parsed = maybeParsed.value
 			if (parsed.method === 'popup_addressBookEntriesChanged') {
-				const chainId = currentRpcNetwork.peek()?.chainId
+				const chainId = currentChain.peek()?.chainId
 				if (chainId !== undefined) sendQuery()
 				return
 			}
 			if (parsed.method === 'popup_settingsUpdated') return sendPopupMessageToBackgroundPage({ method: 'popup_requestSettings' })
 			if (parsed.method === 'popup_requestSettingsReply') {
 				rpcEntries.value = parsed.data.rpcEntries
-				const prevCurrentNetwork = currentRpcNetwork.peek()
+				const prevCurrentNetwork = currentChain.peek()
```

Comment:
`prevCurrentNetwork` ("Previous" "Current") sound opposing. Something like `currentChainSnapshot`, `previousChain` or `previousNetwork` may work

Reply:
the variable being referred is "currentNetwork" so snapshot and chain, chain and network are wrong. I guess `activeNetwork` would be better as we use `activeAddress` as well

Reply:
I added issue for this